tuning/libtuner.py

```diff
@@ -1101,127 +1094,107 @@ def group_benchmark_results_by_device_id(
     return grouped_benchmark_results
 
 
-def parse_grouped_benchmark_results(
-    path_config: PathConfig,
-    grouped_benchmark_results: list[list[TaskResult]],
+def parse_model_benchmark_results(
     candidate_trackers: list[CandidateTracker],
-) -> list[str]:
-    """Update candidate_trackers and collect strings"""
+    candidate_results: list[TaskResult],
+    baseline_results: list[TaskResult],
+):
+    candidate_results = sorted(candidate_results, key=lambda br: br.device_id)
+    baseline_results = sorted(baseline_results, key=lambda tr: tr.device_id)
+
+    # Assign candidates to same groups by device_id
+    grouped_candidate_results = group_benchmark_results_by_device_id(candidate_results)
+
+    # Insert baseline results to the head of each list
+    grouped_benchmark_results = [
+        [x] + y for x, y in zip(baseline_results, grouped_candidate_results)
+    ]
+
     dump_list = []
-    incomplete_list: list[tuple[int, Optional[int]]] = (
+    incomplete_list: list[tuple[int, Optional[str]]] = (
         []
     )  # format: [(candidate_id, device_id)], baseline will have candidate_id=0
 
     for same_device_results in grouped_benchmark_results:
         dump_unsort_list: list[tuple[float, str]] = []
-        for model_candidate_result in same_device_results:
+        for task_result in same_device_results:
             # Skip if benchmark failed.
-            result_str = model_candidate_result.result.stdout
+            result_str = task_result.result.stdout
+            candidate_id = task_result.candidate_id
+            device_id = task_result.device_id
+
             if result_str is None:
+                # TODO: change incomplete process detection
                 continue
 
-            res = ModelBenchmarkResult(result_str)
-            device_id = res.get_device_id()
-
-            # Record baseline benchmarking result.
-            model_candidate_path = res.get_model_candidate_path()
-            if (
-                model_candidate_path is not None
-                and str(path_config.model_baseline_vmfb) in model_candidate_path
-            ):
-                baseline_time = res.get_benchmark_time()
-                if baseline_time is None:
-                    incomplete_list.append((0, device_id))
-                    continue
-                dump_list.append(result_str)
-                continue
+            res = IREEBenchmarkResult(candidate_id, result_str)
+            benchmark_time = res.get_mean_time()
 
-            # Record candidate benchmarking result.
-            c_id = res.get_candidate_id()
-            assert c_id is not None
-            candidate_time = res.get_benchmark_time()
-            if candidate_time is None:
-                incomplete_list.append((c_id, device_id))
+            # Check completion
+            if benchmark_time == None:
```

Comment:
```suggestion
            if benchmark_time is None:
```

This is less ambiguous because it won't use any overloaded operator ==